__tests__/app.test.js

```diff
@@ -65,3 +65,52 @@ describe("GET /api/articles/:article_id", () => {
     expect(response.body.msg).toBe("Bad Request");
   });
 });
+
+describe("GET /api/articles", () => {
+  it("200: responds with all of the articles without body property and with comment_count property", async () => {
+    const response = await request(app).get("/api/articles");
+    expect(response.status).toBe(200);
+    expect(response.body.articles.length).toBe(13);
+    response.body.articles.map((article) => {
+      expect(article).toHaveProperty("author");
+      expect(article).toHaveProperty("title");
+      expect(article).toHaveProperty("article_id");
+      expect(article).toHaveProperty("topic");
+      expect(article).toHaveProperty("created_at");
+      expect(article).toHaveProperty("votes");
+      expect(article).toHaveProperty("article_img_url");
+      expect(article).toHaveProperty("comment_count");
+    });
+    expect(response.body.articles).toBeSortedBy("created_at", {
+      descending: true,
+    });
+  });
+});
+
+describe("GET /api/articles/:article_id/comments", () => {
+  it("200: responds the comments for the requested article ordered by created_at", async () => {
+    const response = await request(app).get("/api/articles/1/comments");
+    expect(response.status).toBe(200);
+    const comments = response.body.comments;
+    expect(comments.length).toBe(11);
+    comments.map((comment) => {
+      expect(comment).toHaveProperty("comment_id");
+      expect(comment).toHaveProperty("body");
+      expect(comment).toHaveProperty("article_id");
+      expect(comment).toHaveProperty("author");
+      expect(comment).toHaveProperty("votes");
+      expect(comment).toHaveProperty("created_at");
+    });
+    expect(comments).toBeSortedBy("created_at");
+  });
+  it("404: responds the appropriate message when article_id is valid but not in the database", async () => {
+    const response = await request(app).get("/api/articles/999/comments");
+    expect(response.status).toBe(404);
+    expect(response.body.msg).toBe("Not Found");
+  });
+  it("400: responds the appropriate message when article_id is not valid", async () => {
+    const response = await request(app).get("/api/articles/invalid/comments");
+    expect(response.status).toBe(400);
+    expect(response.body.msg).toBe("Bad Request");
+  });
```

Comment:
Nice work on this, I think there's one more situation we can handle:
Status: 200 responds with an empty array when the article exists, but contains no comments.
For this one we will need to work out how to differentiate between an empty array due to article not found, and an empty array for a valid article that contains no comments. A checkExists function may help :)